### PR TITLE
Improvement/added flow

### PR DESCRIPTION
**Pull request by rfreitas**
# Problem

We have no type checking in our code base.

# Scope

Get a minimal integration with a type system.

# Solution

Flow. It's not a language but a plugin for babel. It also allow us to activate it only on some files, so we can take a stepped approach to it.

### CODE REVIEW

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-281-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-281-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://281-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-281-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-281-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://281-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-281-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-281-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://281-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by hjheath** [comment]
so what happens if the type is wrong? Do we error or just get a warning?

**Comment by rfreitas** [comment]
you get a lint error.

File: src/components/Camera/index.js [hunk 1]
```diff
@@ -1,3 +1,5 @@
+// @flow
```
Comment (hjheath) [inline src/components/Camera/index.js]: are these necessary? I don't like libraries that use magic comments (ahem [ruby](https://stackoverflow.com/questions/37799296/what-does-the-comment-frozen-string-literal-true-do))

Comment (rfreitas) [reply src/components/Camera/index.js]: necessary. you can remove it if you turn flow for every file I think

Comment (rfreitas) [reply src/components/Camera/index.js]: the `jsx` pgrama mark is necessary though, even though it should be taking it from the babel config.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-281-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-281-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://281-pr-onfido-sdk-ui-onfido.surge.sh

File: src/components/Camera/index.js [hunk 2]
```diff
@@ -31,7 +32,27 @@ const CaptureActions = ({handeClick}) =>
     </button>
   </div>
 
-const CameraPure = ({method, autoCapture, onUploadFallback, onFallbackClick, onUserMedia, faceCaptureClick, countDownRef, webcamRef, onWebcamError}) => (
+type CameraCommonType = {
```
Comment (stefaniacardenas) [inline src/components/Camera/index.js]: In which cases we need to specify the type? Does it need to be done every time that a constant is defined?

Comment (rfreitas) [reply src/components/Camera/index.js]: typing is optional, however for class React components (not pure components), Flow mandates prop types. There is also type inference done by Flow, so even without specifying stuff you can get some type checking for free, but it doesn't seem to work super well for react components.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-281-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-281-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://281-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-281-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-281-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://281-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Some extra changes we might want to include: https://github.com/onfido/onfido-sdk-ui/pull/284/files

File: src/components/Camera/index.js [hunk 3]
```diff
@@ -31,10 +33,29 @@ const CaptureActions = ({handeClick}) =>
     </button>
   </div>
 
+type CameraCommonType = {
+  autoCapture: boolean,
+  method: string,
+  onUserMedia: Function,
+  onUploadFallback: File => void,
+  onWebcamError: Function,
+  autoCapture: boolean,
```
Comment (hjheath) [inline src/components/Camera/index.js]: `autoCapture` is repeated here

**Comment by hjheath** [comment]
I can't get flow to give me an error when I change the types - I think we should look at it together tomorrow.

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-281-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-281-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://staging-281-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://release-281-pr-onfido-sdk-ui-onfido.surge.sh

**Comment by rfreitas** [comment]
Travis automatic deployment: https://281-pr-onfido-sdk-ui-onfido.surge.sh

**Pull request merged**